### ISSUE TITLE
Kape tool commandline bug

### ISSUE THREAD

**Issue by mcvic1rj** [EricZimmerman/KapeFiles#668]
**KAPE version**
V1.2.0.0

**Describe the bug**
When building a command line for a KAPE Module, the paths may contain incorrect quotation marks when KAPE is run from the same disk as the module source

 (note: Removed the duplicate info that I split into a separate issue) 

**To Reproduce**

1. Run KAPE from the Module source disk and use the Magnet Ram Capture module
2. set output directory and module source
    Example CLI: `--msource c: --mdest ..\caps\temp --mflush --module MagnetForensics_RAMCapture --gui`
3. Observe Commandline MRC is run with:
    `Running 'MRC.exe': /accepteula /go "C:\Users\Ryan McVicar\Desktop\caps\temp\Memory"\memory.raw /silent`

Repeat steps above but with KAPE on a different disk and observe the differences in the CLI generated:
`Running 'MRC.exe': /accepteula /go E:\caps\temp\Memory\memory.raw /silent`

**Expected behavior**

Quotes in the path to be after the output filename/path

**Screenshots**

<img width="1710" alt="image" src="https://user-images.githubusercontent.com/10363041/179405728-d7469aa2-5aab-48fd-8320-6f5e8c0b4543.png">

**Additional context**

This may be resolved by v1.3.0 as discussed in #664

**Comment by EricZimmerman** [comment]
What happens if you use an absolute path vs ..

**Comment by mcvic1rj** [comment]
> What happens if you use an absolute path vs ..

Using absolute path functions as expected.

**Comment by EricZimmerman** [comment]
So I guess I need to understand what is wrong here. . means the path where kape is running from. 

Is that not working right now? 

.. is one directory up etc.

**Comment by mcvic1rj** [comment]
In this scenario, it is specifically this bit `"C:\Users\Ryan McVicar\Desktop\caps\temp\Memory"\memory.raw`
MRC crashes due to the way the output path is passed down (first part of path quoted). 

It almost seems like KAPE builds out the `%destinationDirectory%` path, encapsulates it in quotes, then substitutes it in the module for the variable, leading to a path with bad quoting.

**Comment by EricZimmerman** [comment]
Okay. That part will be easy to fix in that I'll just make sure I do all of the substituting and concatenation and then surround it with quotes

**Comment by EricZimmerman** [comment]
so this looks like it can be closed as the other issue that is open relates to the strange quotes thing. im fixing that now. will close this for now too.

@mcvic1rj if you want a beta build to test to verify on your side, just let me know and i can provide in the other thread.

**Comment by EricZimmerman** [comment]
This seems to be working ok? maybe due to the other fix i made:

![image](https://user-images.githubusercontent.com/4265254/179603474-54338019-abba-4c9b-b9ad-1fd5bc659da0.png)


try this binary:

[kapeBETA.zip](https://github.com/EricZimmerman/KapeFiles/files/9135303/kapeBETA.zip)


test your scenarios and report back here if anything is still amiss, with steps to reproduce. its interesting that mrc is a 32 bit executable...

thanks!

**Comment by EricZimmerman** [comment]
report back here with any issues

**Comment by mcvic1rj** [comment]
Looks like that beta build has an issue with the quotation placement when running Kape from the module target drive:
<img width="785" alt="image" src="https://user-images.githubusercontent.com/10363041/179658407-8cc90a74-0433-4d79-8a97-eb524abbb136.png">

**Comment by EricZimmerman** [comment]
you have spaces in the path. ill try that scenario

**Comment by EricZimmerman** [comment]
this is something i cant fix.

the answer is, do not use a path for mdest that has spaces, or use a different memory dumping program, ideally one that has a dedicated switch for the file name to save to. this would fix this whole problem.

tl;dr

set mdest to something WITHOUT spaces, OR if you run it how you are running it, run KAPE from a directory that does NOT contain spaces.

**Comment by mcvic1rj** [comment]
Would it make sense to add a warning in the Kape output when this condition is detected/inside of KapeDocs? I can handle adding a warning to the impacted module’s docs as well
